barretenberg/cpp/src/barretenberg/stdlib/primitives/group/cycle_group.cpp

```diff
@@ -768,70 +752,74 @@ typename cycle_group<Builder>::batch_mul_internal_output cycle_group<Builder>::_
+    BB_ASSERT_EQ(offset_generators.size(), base_points.size() + 1, "Too few offset generators provided!");
     const size_t num_points = scalars.size();
 
+    // Extract the circuit context from any scalar or point
     Builder* context = nullptr;
-    for (auto& scalar : scalars) {
-        if (scalar.lo.get_context() != nullptr) {
-            context = scalar.get_context();
+    for (auto [scalar, point] : zip_view(scalars, base_points)) {
+        if (context = scalar.get_context(); context != nullptr) {
             break;
         }
-    }
-    for (auto& point : base_points) {
-        if (point.get_context() != nullptr) {
-            context = point.get_context();
+        if (context = point.get_context(); context != nullptr) {
             break;
         }
     }
 
+    // Validate all scalars have the same bit length (required for Straus algorithm to process slices)
     size_t num_bits = scalars[0].num_bits();
     for (auto& s : scalars) {
         BB_ASSERT_EQ(s.num_bits(), num_bits, "Scalars of different bit-lengths not supported!");
     }
-    size_t num_rounds = numeric::ceil_div(num_bits, TABLE_BITS);
+    size_t num_rounds = numeric::ceil_div(num_bits, ROM_TABLE_BITS);
 
-    std::vector<straus_scalar_slice> scalar_slices;
+    // Decompose each scalar into 4-bit slices. Note: This operation enforces range constraints on the lo/hi limbs of
+    // each scalar (LO_BITS and (num_bits - LO_BITS) respectively).
+    std::vector<straus_scalar_slices> scalar_slices;
     scalar_slices.reserve(num_points);
     for (const auto& scalar : scalars) {
-        scalar_slices.emplace_back(context, scalar, TABLE_BITS);
+        scalar_slices.emplace_back(context, scalar, ROM_TABLE_BITS);
     }
 
-    /**
-     * Compute the witness values of the batch_mul algorithm natively, as Element types with a Z-coordinate.
-     * We then batch-convert to AffineElement types, and feed these points as "hints" into the cycle_group methods.
-     * This avoids the need to compute modular inversions for every group operation, which dramatically reduces witness
-     * generation times
-     */
+    // Compute the result of each point addition involved in the Straus MSM algorithm natively so they can be used as
+    // "hints" in the in-circuit Straus algorithm. This includes the additions needed to construct the point tables and
+    // those needed to compute the MSM via Straus. Points are computed as Element types with a Z-coordinate then
+    // batch-converted to AffineElement types. This avoids the need to compute modular inversions for every group
+    // operation, which dramatically reduces witness generation times.
     std::vector<Element> operation_transcript;
     Element offset_generator_accumulator = offset_generators[0];
     {
-        // Construct native straus lookup table for each point
+        // For each point, construct native straus lookup table of the form {G , G + [1]P, G + [2]P, ... , G + [15]P}
```

Comment:
just confirming: the native table should be of the form `{G + [1]P, G + [2]P, ..., G + [15]P}`?